ui/components/app/toast-master/selectors.ts

```diff
@@ -86,17 +86,16 @@ export function selectShowConnectAccountToast(
   account: InternalAccount,
 ): boolean {
   const allowShowAccountSetting = getAlertEnabledness(state).unconnectedAccount;
-  const connectedAccounts = getPermittedEVMAccountsForCurrentTab(state);
-  const isEvmAccount = isEvmAccountType(account?.type);
+  const connectedAccounts = getAllPermittedAccountsForCurrentTab(state);
 
-  return (
+  const showConnectAccountToast =
     allowShowAccountSetting &&
     account &&
     state.activeTab?.origin &&
-    isEvmAccount &&
     connectedAccounts.length > 0 &&
-    !connectedAccounts.some((address) => address === account.address)
-  );
+    !connectedAccounts.some((address) => address === account.address);
```

Comment:
i think this is comparing caip-10 account addresses to hex references. I can push a fix for this

Reply:
right. what you said in your screenshot

Reply:
Try this
```
  const showConnectAccountToast =
    allowShowAccountSetting &&
    account &&
    state.activeTab?.origin &&
    !isInternalAccountInPermittedAccountIds(account, connectedAccounts)
```